### PR TITLE
chore: add PyTorchTrial devloper's guide

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

Write down tribal knowledge for a key element of our system in order to properly review PyTorch-native AMP support (#1914)

### CODE REVIEW

File: harness/determined/pytorch/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+
+## Gradient Aggregation:
+
+Gradient aggregation by a factor of N is just like mulitplying the batch size
+by N, except it saves on gpu memory because you process two batches and average
+their gradients before you apply the gradients.  This makes sense in
+distributed training because of the high communication cost of the allreduce.
+(There are some cases where it's not "just like multiplying batch size by N",
+such as with batch norm or if you are TODO:finish)
```
Comment (liamcli) [inline harness/determined/pytorch/README.md:86]: I'm not sure what the remaining TODO is here.

Comment (rb-determined-ai) [reply harness/determined/pytorch/README.md:86]: oh yeah, I think that turned out to be the detail about averaging the aggregated gradients.

**Review by shiyuann — APPROVED** [review]
LGTM
